packages/vite/src/node/plugins/html.ts

```diff
@@ -323,6 +323,20 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
             config,
             publicToRelative,
           )
+        const nodeStartWithLeadingWhitespace = (
+          node: DefaultTreeAdapterMap['node'],
+        ) => {
+          const lineStartOffset =
+            node.sourceCodeLocation!.startOffset -
+            node.sourceCodeLocation!.startCol
+          const line = s.slice(
+            lineStartOffset,
+            node.sourceCodeLocation!.startOffset,
+          )
+          return /\S/.test(line)
+            ? node.sourceCodeLocation!.startOffset
+            : lineStartOffset
+        }
```

Comment:
Name could probably be better, open to suggestions! My brain is failing me at the moment though.

Simply checks if the content preceding the startOffset on the line is white space only, and if so, moves the startOffset to the start of the line so we remove it too.